### PR TITLE
Release 119 - Enable custom HTTP client

### PR DESCRIPTION
**Pull request by tspence**
These patch notes summarize the changes from version 117.0.4438.

Added 5 new APIs:
* ProjectVersion.RetrieveProjectVersions (GET /api/data/projects/{projectId}/versions)
* ProjectVersion.DownloadMSProjectXml (GET /api/data/projects/{projectChangeId}/version/download)
* ProjectVersion.RestoreProjectVersion (POST /api/data/projects/{projectId}/version/{version}/restore)
* ProjectVersion.CopyProjectVersion (POST /api/data/projects/{projectId}/version/{version}/copy)
* Risk.CreateRiskExport (POST /api/data/projects/{projectId}/risks/export)

Renamed 1 old APIs:
* Renamed 'TaskMetadata.GetTasksByProjectIDAndForeignKeyID' to 'TaskMetadata.TaskMetadataSearch'

Changes to data models:
* TaskDto: Added new field `isLocked`
* TaskDto: Added new field `isMilestone`

Enabled construction of a client that uses your own preferred HttpClient object.

### CODE REVIEW

File: src/ProjectManagerClient.cs [hunk 1]
```diff
@@ -333,9 +343,28 @@ public static ProjectManagerClient WithEnvironment(string env, HttpClientHandler
...
+        
+        /// <summary>
+        /// Construct a client that uses a non-standard environment; this can be necessary when using proxy servers or
+        /// an API gateway.  Please be careful when using this mode.
+        /// You should prefer to use `WithEnvironment()` instead wherever possible.
+        /// </summary>
+        /// <param name="env">The named environment to use; should be "production"</param>
+        /// <param name="client">The HttpClient object to use</param>
+        /// <returns>The ProjectManager API client</returns>
+        public static ProjectManagerClient WithCustomHttpClient(string env, HttpClient client)
```
Comment (andrejohansson) [inline src/ProjectManagerClient.cs:359]: Perfect, thanks!

What are your recommendations for test scenarios? To generate trial environment and execute against?

File: src/ProjectManagerClient.cs [hunk 2]
```diff
@@ -249,16 +249,26 @@ public class ProjectManagerClient : IProjectManagerClient
...
+            // If customer wants to use a custom client, let them
+            if (client != null)
+            {
+                _client = client;
+            }
+            else
+            {
+                // Add support for HTTP compression
+                var handler = clientHandler ?? new HttpClientHandler();
+                handler.AutomaticDecompression = DecompressionMethods.GZip;
```
Comment (andrejohansson) [inline src/ProjectManagerClient.cs:265]: Good to know, perhaps this should be a comment on docs when using custom clients (that gzip should be used)?

**Review by andrejohansson — APPROVED** [review]
Thanks for these changes

File: src/ProjectManagerClient.cs (hunk 1 above)
Comment (tspence) [reply src/ProjectManagerClient.cs:359]: We don't have official recommendations; I would defer to your testing team on how they would want to set up mocks.

File: src/ProjectManagerClient.cs (hunk 2 above)
Comment (tspence) [reply src/ProjectManagerClient.cs:265]: Will do

**Review by hermanvos — APPROVED** [review]

**Pull request merged**